### PR TITLE
Allow overwriting of the module command

### PR DESCRIPTION
**Pull request by TrentHouliston**
Allows the generate command to be overwritten for NUbots specific usage

### CODE REVIEW

File: tools/module/generate.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+    # Module help
+    command.help = "Generate a new NUClear Roles module at the provided location"
+
+    # Module subcommands
+    command.add_argument("path", metavar="path", help="a path to the new module (from the module directory)")
+
+
+def run(path, **kwargs):
+    print("hi")
```
Comment (Bidski) [inline tools/module/generate.py:20]: Hi

File: tools/module/generate.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        closeNamespace="\n".join("}" * (len(parts) - 1)),
+    )
+
+
+def generate_cpp(parts):
+    template = textwrap.dedent(
+        """\
+        #include "{className}.h"
+
+        #include "extension/Configuration.h"
```
Comment (Bidski) [inline tools/module/generate.py:130]: This needs to be removed from `nuclear/tools/module/generate.py`

File: tools/module/generate.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+        {openNamespace}
+
+            using extension::Configuration;
```
Comment (Bidski) [inline tools/module/generate.py:134]: This needs to be removed from `nuclear/tools/module/generate.py`

File: tools/module/generate.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+            {className}::{className}(std::unique_ptr<NUClear::Environment> environment)
+            : Reactor(std::move(environment)) {{
+
+                on<Configuration>("{className}.yaml").then([this] (const Configuration& config) {{
+                    // Use configuration here from file {className}.yaml
+                }});
```
Comment (Bidski) [inline tools/module/generate.py:141]: This needs to be removed from `nuclear/tools/module/generate.py`

File: tools/module/generate.py [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+
+def run(path, **kwargs):
+    print("hi")
+    # Try to get our actual module directory from the cmake cache
+    if "NUCLEAR_MODULE_DIR" in b.cmake_cache:
+        module_path = os.path.join(b.source_dir, b.cmake_cache["NUCLEAR_MODULE_DIR"])
+    else:
+        sys.stderr.write("Warning: the system couldn't find the real module directory.")
+        sys.stderr.write("defaulting to module\n")
```
Comment (Bidski) [inline tools/module/generate.py:26]:
```suggestion
        sys.stderr.write("Warning: the system couldn't find the real module directory. Defaulting to module\n")
```

File: tools/module/generate.py (hunk 1 above)
Comment (TrentHouliston) [reply tools/module/generate.py:20]: Hi

File: tools/module/generate.py (hunk 2 above)
Comment (TrentHouliston) [reply tools/module/generate.py:130]: Sure

File: tools/module/generate.py (hunk 3 above)
Comment (TrentHouliston) [reply tools/module/generate.py:134]: Sure

File: tools/module/generate.py (hunk 4 above)
Comment (TrentHouliston) [reply tools/module/generate.py:140]: Sure

File: tools/module/generate.py (hunk 5 above)
Comment (TrentHouliston) [reply tools/module/generate.py]: Just always going with "module"

**Pull request merged**